publisher/internal/analytics/ethereum/types.go

```diff
@@ -18,42 +34,224 @@ type EventLog struct {
 	Removed          bool     `json:"removed"`
 }
 
-type LiquidityAddition struct {
-	repository.LiquidityEntry
-	Token0       TokenAddition
-	Token1       TokenAddition
+type Removal struct {
```

Comment:
It was confusing finding operations implementations inside `types.go`. A better file would be `operations.go` due to the more complex nature of these types.
In other words `types.go` for simple types, and `operations.go` for some complex types that implement interfaces or some more advanced logic.

Thinking outloud:
Maybe Operations are rather generic (all pools contain Add, Remove), so perhaps at least the interface and base implementation could be moved to `pkg/analytics`. But this is up to the developer, because moving it to the `pkg/analytics` will most likely warrant further abstraction that will fall out of scope of this change.